File: problems/linked_lists/removeNode.js
const removeNode = (head, targetVal) => {
  if(head.val === targetVal) return head.next  
  let current = head 
  let prev = null
  while(current) {
    let next = current.next 
    
    if(current.val === targetVal) {
      prev.next = next 
      return head 
      
    }
    prev = current;
    current = current.next;
    
  }
};

File: python_Structy/remove_node.py
class Node:
  def __init__(self, val):
    self.val = val
    self.next = None

def remove_node(head, target_val):
  dummyHead = Node(None)
  dummyHead.next = head
  prev = dummyHead
  next = head.next 
  while head: 
    if head.val == target_val:
      prev.next = next 
      return dummyHead.next
    prev = head 
    head = head.next 
    next = head.next 
    